### DISCUSSION TITLE
Can you use Tailwind with existing libraries that come with their own CSS?

### DISCUSSION THREAD

**Discussion by martinszeltins** [tailwindlabs/tailwindcss#8020] [Help] [answered]
Hi, is there a way to use tailwind with existing js libraries / components that come with their own CSS? Or do you need to override the CSS of those libraries using selectors?

For example, if I am using a jQuery Carousel or maybe a javascript Multiselect library. These libraries render their own HTML so I cannot change the classes and they come with specific styling which I can override like this:

````
.multiselect-container.dropdown-menu {
  background: #dedede;
  border-radius: 4px;
}
````

Does this mean I cannot use Tailwind to style them?

**Comment by bill-kerr** [comment, marked as answer]
You can still target these classes with `@apply`. Check this out:
https://tailwindcss.com/docs/content-configuration#working-with-third-party-libraries

**Reply by martinszeltins** [reply]
> You can still target these classes with `@apply`. Check this out: https://tailwindcss.com/docs/content-configuration#working-with-third-party-libraries

I see. So when working with third party libraries you have to go back to editing a `.css` file with overrides but you still get to use tailwind classes.

**Comment by JibeBigo** [comment]
How would you override the classes from the plugin's css while using a conditional style variable ? 
i.e errorMessage ? 'border-error-600' : 'border-neutral-400

**Reply by bill-kerr** [reply]
Hmm, that's a good question. You could conditionally apply a class on a wrapping element like this:

```jsx
<div class={errorMessage ? 'error' : ''}>
    {plugin}
</div>
```

and then write targeting classes like this:

```css
.plugin-class {
  @apply border-neutral-400;
}

.error .plugin-class {
  @apply border-error-600;
}
```
